File: backend/src/main/java/es/codeurjc/webapp14/dto/ProductDTO.java
package es.codeurjc.webapp14.dto;
import java.util.List;

public record ProductDTO(
    Long id,
    String name,
    String description,
    Double price,
    Integer stock,
    boolean outOfStock,
    int sold,
    String category,
    boolean imageBool,
    String imageUrl,
    List<SizeDTO> sizes,
    List<ReviewDTO> reviews
    //List<OrderProductDTO> orderProducts
) {}

File: frontend/src/app/dtos/product.dto.ts
import { SizeDTO } from "./size.dto";
import { ReviewDTO } from "./review.dto";


export interface ProductDTO {
	id?: number,
	name: string,
	description: string,
    price: number,
    stock: number,
    outOfStock: boolean,
    sold: number,
    category: string,
    imageBool: boolean,
    imageUrl: string,
    reviews: ReviewDTO[]
	sizes: SizeDTO[],
    showDetails?: boolean
}


